Closes #16. Añadida la planificación del proyecto

rocalbbeb opened the pull request:
Se ha añadido un capítulo llamado "Planificación". 

En él se describe la explicación de la metodología seguida en la realización proyecto, y la organización del mismo en Github.

rocalbbeb committed:
Closes #16. Añadido apartado de planificación, que explica la metodología y organización seguida

JJ commented on doc/introduccion.tex:
```diff
@@ -27,4 +27,20 @@ \section{Importancia de la optimización energética en algoritmos evolutivos}
+
+Para ello, se han definido \textbf{historias de usuario} que reflejan las necesidades y expectativas del tribunal en relación con el proyecto. Estas sirven como guía para priorizar y estructurar las tareas, y se han recogido en el repositorio del proyecto mediante \textit{issues} de GitHub. Cada \textit{issue} representa una tarea concreta cuyo cumplimiento satisface una historia de usuario.
+
+El desarrollo se ha organizado en \textbf{hitos} o, en GitHub, \textit{milestones}, que agrupan varias historias de usuario en versiones concretas del proyecto. Cada \textit{milestone} representa un estado alcanzable y verificable del desarrollo, y su descripción refleja el progreso y las funcionalidades incorporadas. 
```
Los hitos no agrupan historias de usuario, sino issues; pero lo más importante es que se trata de un producto mínimamente viable.

JJ reviewed (changes requested):
aparte de lo dicho (y de que falla el workflow) #1 va a requerir que uses una bibliografía adecuada para el proyecto. Sobre todo en estos capítulos, deberías esforzarte por incluir referencias bibliográficas para todo los términos y definiciones que uses.

rocalbbeb committed:
Se han refinado las definiciones #16 y se ha añadido bibliografia #22

rocalbbeb committed:
Se ha añadido bibliografía en el capítulo de planificación #22

rocalbbeb committed:
close #24: Ampliado texto en 'Planificación'

Añadida sección 'Gestión del desarrollo con GitHub'.

JJ commented on doc/introduccion.tex:
```diff
@@ -27,4 +27,24 @@ \section{Importancia de la optimización energética en algoritmos evolutivos}
+
+\chapter{Planificación}
+
+En el presente capítulo se detalla la metodología de desarrollo adoptada durante la creación del proyecto, junto con su control y seguimiento. Todo el proceso ha sido gestionado y documentado en un repositorio de GitHub, donde se han registrado las tareas realizadas, los cambios efectuados y las decisiones tomadas. Extrayendo la información del repositorio, queda reflejado en este capítulo de for ...
```
El hecho de que elijas un repositorio de GitHub forma parte de la metodología, no es un hecho. Dado que los sistemas de desarrollo ágil se basan en uso de sistemas de control de fuentes, puedes ponerlo más adelante cuando *como consecuencia de la metodología usada*, no como un simple hecho. Recuerda que en el TFG lo que hay que hacer es documentar las decisiones.

JJ commented on doc/introduccion.tex:
```diff
@@ -27,4 +27,24 @@ \section{Importancia de la optimización energética en algoritmos evolutivos}
+
+\section{Metodología}
+
+La metodología empleada en este proyecto es la de \textbf{desarrollo ágil}. Esta consiste en realizar de manera iterativa pequeñas unidades de trabajo enfocadas a la satisfacción del cliente. En este caso, el cliente es el tribunal evaluador del Trabajo de Fin de Grado.
```
¿Por qué no pones una cita aquí? No sólo tienes que añadirlas en la bibliografía, también citarlas en el texto.

rocalbbeb replied:
La cita está en el siguiente párrafo, que tambien se ha sacado de la misma fuente. En el libro [Cómo escribir la memoria de tu TFG del Grado en Ingeniería Informática y presentarlo sin morir en el intento](https://digibug.ugr.es/bitstream/handle/10481/102499/C%C3%B3mo%20escribir%20la%20memoria%20de%20un%20TFG%20en%20Inform%C3%A1tica.pdf?sequence=1&isAllowed=y) página 72 indica que para citar en varios párrafos de la misma fuente solo hace falta poner la fuente en el último, por eso lo he hecho así.

JJ replied:
Por un lado, está el tema de que el tribunal tenga presente eso. Por otro lado, está que estás usando unos requisitos que son externos al proyecto y por tanto es imposible que yo los tenga presente a la hora de revisar. A partir de ahí, toma la decisión que veas conveniente.

JJ commented on doc/introduccion.tex:
```diff
@@ -27,4 +27,24 @@ \section{Importancia de la optimización energética en algoritmos evolutivos}
+
+La metodología empleada en este proyecto es la de \textbf{desarrollo ágil}. Esta consiste en realizar de manera iterativa pequeñas unidades de trabajo enfocadas a la satisfacción del cliente. En este caso, el cliente es el tribunal evaluador del Trabajo de Fin de Grado.
+
+El enfoque de desarrollo ágil permite una evolución progresiva del proyecto. De esta manera, se verifica de forma continua que el trabajo realizado se ajusta a los objetivos esperados, lo que facilita detectar a tiempo desviaciones o cambios necesarios. A diferencia de los enfoques tradicionales, en los que la validación suele ocurrir al final del proceso, el desarrollo ágil permite una supervisi ...
```
"objetivos esperados" == "solución del problema del cliente". El desarrollo ágil está centrado en el cliente. No puedes hablar de "objetivos" en abstracto, porque siempre van a estar relacionadas con el cliente.

JJ commented on doc/introduccion.tex:
```diff
@@ -27,4 +27,24 @@ \section{Importancia de la optimización energética en algoritmos evolutivos}
+
+\section{Gestión del desarrollo con GitHub}
+
+Para seguir las pautas de la metodología, se han definido \textbf{historias de usuario}, que reflejan las necesidades y expectativas del tribunal en relación con el proyecto. Estas sirven como guía para priorizar y estructurar las tareas, y se han recogido en el repositorio del proyecto mediante \textit{issues} de GitHub. Cada \textit{issue} representa una tarea concreta cuyo cumplimiento satisfa ...
```
No del tribunal, *del cliente*

JJ commented on doc/introduccion.tex:
```diff
@@ -27,4 +27,24 @@ \section{Importancia de la optimización energética en algoritmos evolutivos}
+
+Para seguir las pautas de la metodología, se han definido \textbf{historias de usuario}, que reflejan las necesidades y expectativas del tribunal en relación con el proyecto. Estas sirven como guía para priorizar y estructurar las tareas, y se han recogido en el repositorio del proyecto mediante \textit{issues} de GitHub. Cada \textit{issue} representa una tarea concreta cuyo cumplimiento satisfa ...
+
+La gestión del flujo de trabajo se ha llevado a cabo mediante un tablero \textbf{Kanban}, integrado en GitHub Projects, que permite visualizar el estado de cada tarea (pendiente, en progreso, completada) y facilita el seguimiento continuo del proyecto \cite{GitHubProjects2024}. 
```
Si has usado un tablero Kanban, haz una captura de pantalla del mismo.

JJ commented on doc/introduccion.tex:
```diff
@@ -27,4 +27,24 @@ \section{Importancia de la optimización energética en algoritmos evolutivos}
+
+El desarrollo se ha organizado en \textbf{hitos} o, en GitHub, \textit{milestones}, que agrupan varias \textit{issues} en versiones concretas del proyecto. Un \textit{milestone} se trata de un producto mínimamente viable, y representa un estado alcanzable y verificable del desarrollo \cite{GitHubMilestones2024}.
+
+La documentación mediante un repositorio de GitHub permite un seguimiento claro del avance del proyecto, así como mantener una organización en el mismo. Además, las \textit{issues} y los \textit{milestones} ayudan a asegurar y verificar que los requisitos definidos se corresponden con el trabajo realizado. 
```
La planificación inicial del proyecto requiere un par de productos mínimamente viables. ¿Lo vas a incluir en este PR o con esto ya estaría terminado?

JJ reviewed (commented):
¿No te parece que una introducción debe ser solo una introducción? ¿No tiene la metodología entidad suficiente para su propio capítulo?

rocalbbeb committed:
Incluido planificacion.tex en el Makefile #25

Para que el proyecto compile con cada cambio en planificacion.tex

rocalbbeb committed:
Separada planificación de introduccion.tex #25

Se ha creado un archivo separado para la planificación, quitandola del archivo de introducción

rocalbbeb committed:
Incluido planificacion.tex en el archivo memoria.tex #25

Para que la planificación se añada a la memoria, se ha añadido en el archivo principal de la memoria

rocalbbeb committed:
Closes #25. Añadido archivo planificacion.tex de la memoria

Se ha creado un archivo para el capítulo de planificación de la memoria

rocalbbeb committed:
Advertencias de textidote resueltas #23

Añadido identificador de una url, oprimida la label en la imagen de kanban porque no se usa y reorganizada la sección de organización del proyecto, que ha pasado a llamarse organización en milestones porque se ha oprimido la subsección de milestones

rocalbbeb committed:
Advertencias de textidote resueltas #23

Se ha agregado label a la imagen de kanban para referenciarla en el texto y se ha añadido más texto a la sección de organización en milestones

JJ commented on doc/bibliografia.bib:
```diff
@@ -113,4 +113,29 @@ @article{jamil2022analyzing
+  title        = {Planning and tracking with projects},
+  year         = {2024},
+  url          = {https://docs.github.com/en/issues/planning-and-tracking-with-projects},
+  note         = {Recuperado de \url{https://docs.github.com/en/issues/planning-and-tracking-with-projects}}
```
Normalmente en las web se pone en qué fecha se han consultado. Ya está el URL en url de toas formas, no tienes qeu repetirlo aquí.

JJ commented on doc/introduccion.tex:
```diff
@@ -4,15 +4,15 @@ \chapter{Introducción}
-La inteligencia computacional es una rama de la inteligencia artificial que se centra en desarrollar sistemas inteligentes inspirados en la naturaleza y el razonamiento humano. Entre sus principales enfoques se encuentran las redes neuronales artificiales, que imitan el funcionamiento del sistema nervioso, y los algoritmos de optimización bioinspirados, como los algoritmos evolutivos, que constit ...
+La inteligencia computacional es una rama de la inteligencia artificial que se centra en desarrollar sistemas inteligentes inspirados en la naturaleza y el razonamiento humano. Entre sus principales enfoques se encuentran las redes neuronales artificiales, que imitan el funcionamiento del sistema nervioso, y los algoritmos de optimización bioinspirados, como los algoritmos evolutivos, que constit ...
 
 La inteligencia artificial (IA) ha experimentado un crecimiento exponencial en los últimos años, lo que ha provocado importantes implicaciones medioambientales. Según estimaciones recientes de 2023 reflejadas en \cite{cotta2024consumo}, los centros de datos dedicados al procesamiento de IA representan entre el 5\% y el 9\% de la demanda mundial de electricidad y generan aproximadamente el 2\% de  ...
```
No sé si merece la pena que hables de esto. Los algoritmos evolutivos son difícilmente IA

JJ commented on doc/introduccion.tex:
```diff
@@ -4,15 +4,15 @@ \chapter{Introducción}
 A medida que el sector de la inteligencia artificial avanza a un ritmo acelerado, las leyes y regulaciones destinadas a controlar su impacto medioambiental evolucionan de forma más lenta. Como consecuencia, muchos proyectos de IA no aplican criterios de sostenibilidad en su desarrollo. A esta tendencia de creación y uso de tecnologías de IA sin considerar su huella ambiental se la conoce como \te ...
 
-Por el contrario, se denomina \textbf{Green Computing} o \textbf{IA verde} a aquellas prácticas de computación que integran consideraciones medioambientales en su desarrollo, con el objetivo de alcanzar un equilibrio entre el rendimiento computacional y el coste energético. En este enfoque, no se persigue únicamente el avance tecnológico, sino que también se prioriza la sostenibilidad durante tod ...
+Por el contrario, se denomina \textbf{Green Computing} o \textbf{IA verde} a aquellas prácticas de computación que integran consideraciones medioambientales en su desarrollo, con el objetivo de alcanzar un equilibrio entre el rendimiento computacional y el coste energético. En este enfoque, no se persigue únicamente el avance tecnológico, sino que también se prioriza la sostenibilidad durante tod ...
```
Quizás IC verde, no IA verde.

JJ commented on doc/planificacion.tex:
```diff
@@ -0,0 +1,45 @@
+
+\section{Gestión del desarrollo con GitHub}
+
+Como consecuencia de adoptar la metodología de desarrollo ágil, el desarrollo del proyecto debe basarse en un sistema de control. Se ha optado por usar un repositorio \textbf{GitHub} para documentar y seguir el proceso del proyecto. 
```
Las decisiones no se enuncian, se justifican. Una aplicación es una secuencia de decisiones que son respuesta a problemas claros y bien planteados.

JJ commented on doc/planificacion.tex:
```diff
@@ -0,0 +1,45 @@
+
+Como consecuencia de adoptar la metodología de desarrollo ágil, el desarrollo del proyecto debe basarse en un sistema de control. Se ha optado por usar un repositorio \textbf{GitHub} para documentar y seguir el proceso del proyecto. 
+
+Para seguir las pautas de la metodología, se han definido \textbf{historias de usuario}, que reflejan las necesidades y expectativas del cliente en relación con el proyecto. Estas sirven como guía para priorizar y estructurar las tareas, y se han recogido en el repositorio del proyecto mediante \textit{issues} de GitHub. Cada \textit{issue} representa una tarea concreta cuyo cumplimiento satisfac ...
```
¿Qué metodología? REcuerda que ágil no es una metodología, es una mentalidad o  un enfoque, y para cada cosa hay que decir qué metodología se ha elegido. Lo cierto es que no sé si el usar historias de usuario es una metodología específica, así que en ese caso simplemetne no menciones una metodología, sino di qué es lo que quiere el desarrollo ágil y como lo vas a llevar a cabo.

JJ commented on doc/planificacion.tex:
```diff
@@ -0,0 +1,45 @@
+
+Para seguir las pautas de la metodología, se han definido \textbf{historias de usuario}, que reflejan las necesidades y expectativas del cliente en relación con el proyecto. Estas sirven como guía para priorizar y estructurar las tareas, y se han recogido en el repositorio del proyecto mediante \textit{issues} de GitHub. Cada \textit{issue} representa una tarea concreta cuyo cumplimiento satisfac ...
+
+La gestión del flujo de trabajo se ha llevado a cabo mediante un tablero \textbf{Kanban}, que se muestra en la Figura ~\ref{fig:kanban}. Integrado en GitHub Projects, permite visualizar el estado de cada tarea (pendiente, en progreso, en revisión y completada) y facilita el seguimiento continuo del proyecto \cite{GitHubProjects2024}. 
```
Esto *si* es una metodología para la secuenciación de las tareas y su agrupación en sprints al final del cual hay un producto mínimamente viable. Como tal metodología, es también una decisión *que tienes que documentar*

JJ reviewed (changes requested):
Los principales problemas es que tienes *siempre* que justificar las decisiones que tomes, explicando qué es lo que que requieres de la herramienta o metodología que vas a elegir, qué opciones hay y de qué forma cumplen o no las expectativas. Con esto demuestras conocimiento del estado del arte, pero también madurez a la hora de tomar y justificar decisiones.
